go/pkg/vm/description.go

```diff
@@ -63,27 +112,40 @@ func (self *Description) QemuCommandLine() ([]string, error) {
 		strconv.FormatUint(uint64(self.memory), 10),
 	}
 
-	if self.cpus == 0 {
-		return ret, errors.New("number of CPUs not set")
+	err := self.validate()
+	if err != nil {
+		return ret, err
 	}
 
-	if self.memory == 0 {
-		return ret, errors.New("memory size not set")
+	switch self.model {
+	case LegacyV1:
```

Comment:
I think a good way to handle this, is keeping the information, but adding "legacy" or "deprecated". Something like: `MachineModelPCLegazy` or `MachineModelPCDeprecated`.

An unimportant side-note: A nice thing about adding `Deprecated` later on to constants is, that people can still use the old things, but when they vendor in new code, the code does not initially compile, which makes them aware of the change.

A change I would want to see: In general it is common to have constant with a prefix which indicates the meaning. In this case `MachineModelQ35` and `MachineModelPCLegacy`.

Reply:
> I think a good way to handle this, is keeping the information, but adding "legacy" or "deprecated". Something like: `MachineModelPCLegazy` or `MachineModelPCDeprecated`.
> 
> An unimportant side-note: A nice thing about adding `Deprecated` later on to constants is, that people can still use the old things, but when they vendor in new code, the code does not initially compile, which makes them aware of the change.

That's an interesting approach! I'll think a bit more about it, but it sounds perfectly viable at first glance.

> A change I would want to see: In general it is common to have constant with a prefix which indicates the meaning. In this case `MachineModelQ35` and `MachineModelPCLegacy`.

That makes sense, and in fact if you look at the C bindings that's pretty much exactly how I've done it. I guess I kinda got confused with Rust, where you'd use it as `Model::LegacyV1` and thus cause no confusion O:-)

The reason I didn't use the names `pc` and `q35` in the constants is that, when choosing the `vm.Model`, the user is not just locking into a machine type but also into a PCI address allocation algorithm and potentially even more stuff in the future, so I thought something slightly more abstract would be appropriate. Perhaps though we can be more explicit and rely on the versioning for the details?

For example, `VmModelPcV1` might mean "`pc-4.2` machine type with all devices on `pci.0`" whereas `VmModelPcV2` might mean "`pc-5.0` machine with devices on separate `pci-pxb`" and so on.

Reply:
> The reason I didn't use the names `pc` and `q35` in the constants is that, when choosing the `vm.Model`, the user is not just locking into a machine type but also into a PCI address allocation algorithm and potentially even more stuff in the future, so I thought something slightly more abstract would be appropriate.

Would users of virtblocks still be able to select a specific machine type, or do you expect a strong relationship between the algorithm, the machine type and the version?

Reply:
> Would users of virtblocks still be able to select a specific machine type, or do you expect a strong relationship between the algorithm, the machine type and the version?

And as a follow-up thought, would it make sense to explicitly select an algorithm, orthogonal to the machine type?

Just for illustration, something like:
```go
.Machine(MachineModelQ3542).AllocationAlgorithm(AllocationAlgorithmV1)
```

Reply:
The level of granularity we want is still up for debate, and the correct balance will only be found through experimentation I think. We definitely need to make it possible to use arbitrary machine types, if nothing else for libvirt's sake.

The original idea behind VM models was that they would represent a more fully-formed VM than QEMU's machine types, but since formulating it we've already somewhat revisited it in that, for example, you don't automatically get any disk slots or serial consoles but have to add them manually instead.

So with that in mind, setting machine type and PCI allocation algorithm separately could make sense, or at the very least I don't see any obvious problems with it right now :)

Would you consider having the two tied together, as it is in this PR, as a deal-breaker? Because, if you don't, I think it would be okay to merge this version after fixing the naming for `vm.Model`, with the understanding that the main reason we didn't split them yet is to avoid spending more unallocated time on this and we can easily make the split happen later if we decide to invest more resources on Virt Blocks.

Reply:
> Would you consider having the two tied together, as it is in this PR, as a deal-breaker? Because, if you don't, I think it would be okay to merge this version after fixing the naming for `vm.Model`, with the understanding that the main reason we didn't split them yet is to avoid spending more unallocated time on this and we can easily make the split happen later if we decide to invest more resources on Virt Blocks.

Works for me.